src/FixedPointNumbers.jl

```diff
@@ -42,25 +40,198 @@ export
+reinterpret{T,f}(::Type{T}, x::FixedPoint{T,f}) = x.i
+reinterpret{T <: Integer,f}(::Type{FixedPoint{T,f}}, x::T) = FixedPoint{T,f}(x, 0)
+
+rawtype{T,f}(::Type{FixedPoint{T,f}}) = T
+nbitsfrac{T,f}(::Type{FixedPoint{T,f}}) = f
+rawone(v) = reinterpret(one(v))
 
-# comparison
-=={T <: FixedPoint}(x::T, y::T) = x.i == y.i
- <{T <: FixedPoint}(x::T, y::T) = x.i  < y.i
-<={T <: FixedPoint}(x::T, y::T) = x.i <= y.i
+# comparisons
+=={T <: FixedPoint}(x::T, y::T) = reinterpret(x) == reinterpret(y)
+ <{T <: FixedPoint}(x::T, y::T) = reinterpret(x)  < reinterpret(y)
+<={T <: FixedPoint}(x::T, y::T) = reinterpret(x) <= reinterpret(y)
 
 # predicates
 isinteger{T,f}(x::FixedPoint{T,f}) = (x.i&(1<<f-1)) == 0
+isfinite(x::FixedPoint) = true
+isnan(x::FixedPoint) = false
+isinf(x::FixedPoint) = false
 
+# traits
 typemax{T<: FixedPoint}(::Type{T}) = T(typemax(rawtype(T)), 0)
 typemin{T<: FixedPoint}(::Type{T}) = T(typemin(rawtype(T)), 0)
 realmin{T<: FixedPoint}(::Type{T}) = typemin(T)
 realmax{T<: FixedPoint}(::Type{T}) = typemax(T)
+eps{T<:FixedPoint}(::Type{T}) = T(one(rawtype(T)),0)
+eps{T<:FixedPoint}(::T) = eps(T)
+sizeof{T<:FixedPoint}(::Type{T}) = sizeof(rawtype(T))
 
-include("fixed.jl")
-include("ufixed.jl")
-include("deprecations.jl")
+zero{T <: FixedPoint}(::Type{T}) = T(zero(rawtype(T)),0)
+ one{T <: FixedPoint}(::Type{T}) = T(2^nbitsfrac(T)-1,0)
+zero(x::FixedPoint) = zero(typeof(x))
+ one(x::FixedPoint) =  one(typeof(x))
+
+
+# basic operators & arithmetics
+(-){T<:FixedPoint}(x::T) = T(-reinterpret(x), 0)
+(~){T<:FixedPoint}(x::T) = T(~reinterpret(x), 0)
+abs{T<:FixedPoint}(x::T) = T(abs(reinterpret(x.i)),0)
+
++{T<:FixedPoint}(x::T, y::T) = T(reinterpret(x) + reinterpret(y),0)
+-{T<:FixedPoint}(x::T, y::T) = T(reinterpret(x) - reinterpret(y),0)
+
+# with truncation:
+# *{T<:FixedPoint}(x::T, y::T) =
+#         T(Base.widemul(reinterpret(x),reinterpret(y)) >> nbitsfrac(T),0)
+# with rounding up:
+function *{T<:FixedPoint}(x::T, y::T)
+    f = nbitsfrac(T)
+    i = Base.widemul(reinterpret(x),reinterpret(y))
+    T((i + convert(widen(rawtype(T)), 1) << (f-1) )>>f,0)
+end
+
+function /{T<:FixedPoint}(x::T, y::T)
+    f = nbitsfrac(T)
+    T(div(convert(widen(rawtype(T)), x.i) << f, y.i), 0)
+end
+
+# Conversions to FixedPoint
+convert{T,f}(::Type{FixedPoint{T,f}}, x::Integer) =
+    FixedPoint{T,f}(convert(T,x)<<f,0)
+convert{T,f}(::Type{FixedPoint{T,f}}, x::AbstractFloat) =
+    FixedPoint{T,f}(trunc(T,x) << f + round(T, rem(x,1)*(1<<f)),0)
+convert{T,f}(::Type{Fixed{T,f}}, x::Rational) =
+    FixedPoint{T,f}(x.num)/FixedPoint{T,f}(x.den)
+
+# Conversions from FixedPoint
+convert{T,f}(::Type{BigFloat}, x::FixedPoint{T,f}) =
+    convert(BigFloat,x.i>>f) + convert(BigFloat,x.i&(1<<f - 1))/convert(BigFloat,1<<f)
+# convert(::Type{BigFloat}, x::UFixed) = reinterpret(x)*(1/BigFloat(rawone(x)))
+
+convert{TF<:AbstractFloat,T,f}(::Type{TF}, x::FixedPoint{T,f}) =
+    convert(TF,x.i>>f) + convert(TF,x.i&(1<<f - 1))/convert(TF,1<<f)
```

Comment:
I think the division will be slower than the former multiplication. Also, try this on `x.i = 0xff`: you get 255/256, which is not quite 1.